receiver/splunkhecreceiver/receiver.go

```diff
@@ -295,18 +378,44 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, err)
 		return
 	}
+
 	consumerErr := r.logsConsumer.ConsumeLogs(ctx, ld)
 
 	_ = bodyReader.Close()
 
 	if consumerErr != nil {
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, consumerErr)
 	} else {
-		r.writeSuccessResponse(ctx, resp, ld.LogRecordCount())
+		if len(partitionID) > 0 && r.ackExt != nil {
```

Comment:
What does it mean when the partition ID is <= 0? (Why do we not want to send an ack in that case?)

Reply:
if partitionID > 0 && r.ackExt, that means the ackextension is enabled and the request contains a partition ID, indicating that it wants to ack for this particular event being ingested by hec receiver. In which case, the `writeSuccessResponseWithAck` will send back a ackID associated with the event which the client can query later to see whether if has been acked or not. If the condition is false, then we just send back a standard HEC data received message without the ackID.